new module for google cloud plateform

GwenaelPellenArkeup opened the pull request:
The module can manage Instance Templates of Google Compute Engine.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
gce_template

##### ANSIBLE VERSION
```
ansible 2.2.1.0
```

##### SUMMARY
...

GwenaelPellenArkeup committed:
new module for google cloud plateform

The module can manage Instance Templates of Google Compute Engine.

jasperla commented on lib/ansible/modules/cloud/google/gce_template.py:
```diff
@@ -0,0 +1,565 @@
+    required: false
+    default: "present"
+    choices: ["present", "absent"]
+    aliases: []
```
There is no need to list empty aliases.

jasperla commented on lib/ansible/modules/cloud/google/gce_template.py:
```diff
@@ -0,0 +1,565 @@
+    - "apache-libcloud >= 0.13.3, >= 0.17.0 if using JSON credentials,
+      >= 0.20.0 if using preemptible option"
+notes:
+  - Either I(image) or I(image_family) is required.
```
You can use `required_one_of=[['image', 'image_family']]` as an argument to `AnsibleModule` to enforce this.

jasperla commented on lib/ansible/modules/cloud/google/gce_template.py:
```diff
@@ -0,0 +1,565 @@
+
+import socket
+from ansible.module_utils.basic import *
+from ansible.module_utils.gce import *
```
Please only import what you actually use instead of `*`.

jasperla commented on lib/ansible/modules/cloud/google/gce_template.py:
```diff
@@ -0,0 +1,565 @@
+
+
+def main():
+    module = AnsibleModule(
```
Would it be possible to add support for `supports_check_mode`?

GwenaelPellenArkeup committed:
new module for google cloud plateform

remove empty aliases (not need)

GwenaelPellenArkeup committed:
new module for google cloud plateform

fix #pullrequestreview-18841338 : add require_one_of parameter.

GwenaelPellenArkeup committed:
new module for google cloud plateform

fix #pullrequestreview-18841404 : change import
fix #pullrequestreview-18841625 : create check_if_system_state_would_be_changed and add parameter supports_check_mode.

GwenaelPellenArkeup committed:
new module for google cloud plateform

fix #pullrequestreview-18841404 : change import
fix #pullrequestreview-18841625 : create check_if_system_state_would_be_changed and add parameter supports_check_mode.
new module for google cloud plateform

fix #pullrequestreview-18841338 : add require_one_of parameter.
new module for google cloud plateform

remove empty aliases (not need)
new module for google cloud plateform

The module can manage Instance Templates of Google Compute Engine.

GwenaelPellenArkeup committed:
Merged branch devel into devel

GwenaelPellenArkeup committed:
Merge remote-tracking branch 'refs/remotes/ansible/devel' into devel

mattclay commented:
Closing and re-opening to trigger CI.

GwenaelPellenArkeup committed:
Merge remote-tracking branch 'refs/remotes/ansible/devel' into devel

GwenaelPellenArkeup commented:
@walbert947 @supertom @dohoangkhiemgmail.com need_shipit
